new_webapp/src/components/TopBar.js

```diff
@@ -13,4 +15,24 @@ export const TopBar = () => (
 
 const flavorText = `
 Hoogle+ is a type-driven synthesis engine for Haskell - like Hoogle but able to find compositions of functions. Given a Haskell type, Hoogle+ generates terms that inhabit this type by composing library components. It supports polymorphism, type classes, and higher-order functions. Available library components are listed in the side bar.
-`;
\ No newline at end of file
+`;
+
+const dlStyle = {position:"absolute",
+   top:0,
+   right:0
+}
+
+function dlClick(e) {
+	var element = document.createElement('a');
+	var msgs = log.exportToArray();
+	var text = msgs.join('\n');
+    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
+    element.setAttribute('download', 'hoogle-plus.log');
+
+    element.style.display = 'none';
+    document.body.appendChild(element);
+
+    element.click();
+
+    document.body.removeChild(element);
```

Comment:
I'm a little surprised react doesn't barf on you for changing the dom, but if it works, gr8.

Reply:
Actually I only tested this in FF, if someone can check it out and try it on Chrome that'd be good.